lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -391,6 +415,7 @@ def wrap_init(self: CharmBase, framework: Framework, *args, **kwargs):
         _service_name = service_name or f"{self.app.name}-charm"
 
         unit_name = self.unit.name
+        _remove_stale_otel_sdk_packages()
```

Comment:
I'm not super convinced it's enough to do it here, perhaps we should apply the patch on module import. Only a problem if charmers use the otlp SDK to set up their own tracers before the charm inits (which is a very edge case), so for now let's choose to ignore it